src/index.jsx

```diff
@@ -34,12 +34,36 @@ const apiClient = getAuthenticatedAPIClient({
   loggingService: NewRelicLoggingService,
 });
 
+let callbackAccessToken = null;
+
+/**
+ * Temporary check of auth state vs callback access token to help debug and fix any
+ * potential problems.
+ */
+function checkAuthState(functionName, authenticationState) {
+  const userIdsMatch = !!(
+    callbackAccessToken && callbackAccessToken.user_id &&
+    authenticationState && authenticationState.authentication &&
+    authenticationState.authentication.userId &&
+    callbackAccessToken.user_id === authenticationState.authentication.userId
+  );
+  const customAttributes = {
+    functionName,
+    callbackAccessToken,
+    authenticationState,
+    userIdsMatch,
+  };
+  NewRelicLoggingService.logInfo(`checkAuthState: ${functionName}: userIdsMatch=${userIdsMatch}`, customAttributes);
+}
+
 /**
  * We need to merge the application configuration with the authentication state
  * so that we can hand it all to the redux store's initializer.
  */
 function createInitialState() {
-  return Object.assign({}, { configuration }, apiClient.getAuthenticationState());
+  const authenticationState = apiClient.getAuthenticationState();
+  checkAuthState('createInitialState', authenticationState);
```

Comment:
Note to reviewers: if the state doesn't match in certain cases, there is probably a quick workaround solution to use the cached callbackAccessToken instead, and we can invest more time in fixing frontend-auth later.